instrumentation/dalli/test/opentelemetry/instrumentation/dalli/instrumentation_test.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+require_relative '../../../../lib/opentelemetry/instrumentation/dalli'
+require_relative '../../../../lib/opentelemetry/instrumentation/dalli/patches/server'
+
+describe OpenTelemetry::Instrumentation::Dalli::Instrumentation do
+  let(:instrumentation) { OpenTelemetry::Instrumentation::Dalli::Instrumentation.instance }
+  let(:exporter) { EXPORTER }
+  let(:span) { exporter.finished_spans.first }
+  let(:host) { ENV.fetch('TEST_MEMCACHED_HOST') { '127.0.0.1' } }
+  let(:port) { ENV.fetch('TEST_MEMCACHED_PORT') { 11211 } }
+  let(:dalli) { ::Dalli::Client.new("#{host}:#{port}", {}) }
+
+  before do
+    exporter.reset
+  end
+
+  after do
+    # Force re-install of instrumentation
+    instrumentation.instance_variable_set(:@installed, false)
+  end
+
+  describe 'tracing' do
+    before do
+      instrumentation.install
+    end
+
+    it 'before request' do
+      _(exporter.finished_spans.size).must_equal 0
+    end
+
+    it 'after dalli#set' do
+      dalli.set('foo', 'bar')
+
+      _(exporter.finished_spans.size).must_equal 1
+      _(span.name).must_equal 'set'
+      _(span.attributes['db.system']).must_equal 'memcached'
+      _(span.attributes['db.statement']).must_equal 'set foo bar 0 0'
+      _(span.attributes['net.peer.name']).must_equal host
+      _(span.attributes['net.peer.port']).must_equal port
+    end
+
+    it 'after dalli#set' do
+      dalli.get('foo')
+
+      _(exporter.finished_spans.size).must_equal 1
+      _(span.name).must_equal 'get'
+      _(span.attributes['db.system']).must_equal 'memcached'
+      _(span.attributes['db.statement']).must_equal 'get foo'
+      _(span.attributes['net.peer.name']).must_equal host
+      _(span.attributes['net.peer.port']).must_equal port
+    end
+
+
+    it 'after dalli#get_multi' do
+      dalli.get_multi('foo', 'bar')
+
+      _(exporter.finished_spans.size).must_equal 1
+      _(span.name).must_equal 'getkq'
+      _(span.attributes['db.system']).must_equal 'memcached'
+      _(span.attributes['db.statement']).must_equal 'getkq foo bar'
+      _(span.attributes['net.peer.name']).must_equal host
+      _(span.attributes['net.peer.port']).must_equal port
+    end
+
+    it 'after error' do
+      dalli.set('foo', 'bar')
+      exporter.reset
+
+      dalli.instance_variable_get(:@ring).servers.first.stub(:write, ->(_bytes) { raise Dalli::NetworkError }) do
```

Comment:
I'm calling `.set` so the client sets up a server in the ring instance variable, then grabbing it for a stub. I want to try and simulate an actual error that could crop up.  There wasn't a better way obvious to me on how to do this.

Reply:
lgtm, i also have no ideas here off the top of my head

Reply:
Is there an equivalent of `Dalli::Server.any.stub(...)` to avoid reaching in to the instance variable?

Otherwise this seems fine to me. You could go to the trouble of pulling in Semian, but I'm not sure it's worth it yet.

Reply:
Hmm, doesn't look like Semian would help here - it doesn't include support for Dalli AFAICT.

Reply:
>Is there an equivalent of Dalli::Server.any.stub(...) to avoid reaching in to the instance variable?

I've been looking, and it seems minitest doesn't have this baked in and I didn't really want to pull in another test dependency for just this one test.